projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/credentials/AWSServiceAccountCredentialsService.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2023-2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.service.credentials;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.BasicSessionCredentials;
+import com.amazonaws.auth.STSAssumeRoleSessionCredentialsProvider;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenService;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClientBuilder;
+import com.amazonaws.services.securitytoken.model.AssumeRoleRequest;
+import java.util.UUID;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AWSServiceAccountCredentialsService {
+
+  @Value("${datajobs.aws.serviceAccountSecretAccessKey:}")
+  private String awsSecretAccessKey;
+
+  @Value("${datajobs.aws.serviceAccountAccessKeyId:}")
+  private String awsAccessKeyId;
+
+  @Value("${datajobs.aws.RoleArn:}")
+  private String awsRoleArn;
+
+  @Value("${datajobs.aws.defaultSessionDurationSeconds:}")
+  private int sessionDuration;
+
+  @Value("${datajobs.aws.region:}")
+  private String awsRegion;
+
+  public BasicSessionCredentials getTemporaryCredentials() {
```

Comment:
The name of the getTemporaryCredentials method implies that it returns the same set of credentials each time it is called. However, the method actually returns a new set of temporary credentials each time it is called. Might be better to rename the method to createTemporaryCredentials or similar.